src/vs/workbench/contrib/terminal/common/terminalEnvironment.ts

```diff
@@ -235,6 +235,17 @@ export function createTerminalEnvironment(
 			if (shellLaunchConfig.env) {
 				resolveConfigurationVariables(configurationResolverService, shellLaunchConfig.env, lastActiveWorkspace);
 			}
+			if (shellLaunchConfig.args) {
```

Comment:
Can you do this for `.executable` as well? That maps to `terminal.integrated.shell.<platform>`, this should be done in `getDefaultShell`.